### PR TITLE
fix indentation in ternaries nested in conditions of other ternaries

### LINKED ISSUE #5257: invalid indent with nested ternary in test
**Prettier 1.14.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEASAbgQwE4AIC8uAFKlAK4C2ARnHgKS4CMADLgHyEBMuAZDwDpRcw4qUo16TZqwA8U3AB8FuMdVq4GLVh1ydmASkEjjuAPxMjJ4Ul2GhI880vXczANwgANCAgAHGACW0ADOyKA42BAA7gAKOAihKJjoEAEAJl4gVNiYYADWcDAAyr65AVAA5sgw2GRw3uXBtDAxORUUmMgAZpgANk3eAFbBAB4AQjn5hUWYFHAAMuVw3X0DIKXYTdjIWZhUAJ690Jm+2OUwAOrpMAAWyAAczN6nEE0XOb47p3Bb6Mve2DgAEcyAFAa1MO1Okgev16iAmhQAtVavDguUKr04ABFMgQeArOHeGB7K5pW7ITjEnIBXoYgDCEAoHR2UGg-xAZCaABU9olYU0AL6CoA)
```sh
--parser babylon
```

**Input:**
```jsx
$var = ($number % 10 >= 2 &&
    ($number % 100 < 10 || $number % 100 >= 20)
        ? 1
        : 2)
    ? 0
    : 0;
```

**Output:**
```jsx
$var = ($number % 10 >= 2 && ($number % 100 < 10 || $number % 100 >= 20)
? 1
: 2)
  ? 0
  : 0;

```

**Expected behavior:**

```jsx
$var = ($number % 10 >= 2 && ($number % 100 < 10 || $number % 100 >= 20)
  ? 1
  : 2)
    ? 0
    : 0;

```

Maybe even better output, hard to say what will be looks good, but indent should be don't loose.

### PR DESCRIPTION
**Pull request by thorn0**
fixes #5257

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/ternaries/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -2424,6 +2424,350 @@ a
...
+  </Element2>
+);
+
+=====================================output=====================================
+$var = ($number % 10 >= 2 && ($number % 100 < 10 || $number % 100 >= 20)
+  ? kochabCooieGameOnOboleUnweave
+  : annularCooeedSplicesWalksWayWay
+)
+  ? anodyneCondosMalateOverateRetinol
+  : averredBathersBoxroomBuggyNurl;
```
Comment (thorn0) [inline tests/ternaries/__snapshots__/jsfmt.spec.js.snap]: Wouldn't the following formatting look better here I wonder?
```js
$var = (
  $number % 10 >= 2 && ($number % 100 < 10 || $number % 100 >= 20)
    ? kochabCooieGameOnOboleUnweave
    : annularCooeedSplicesWalksWayWay
)
  ? anodyneCondosMalateOverateRetinol
  : averredBathersBoxroomBuggyNurl;
```

Comment (evilebottnawi) [reply tests/ternaries/__snapshots__/jsfmt.spec.js.snap]: I think your example better :+1:

File: tests/ternaries/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -2424,6 +2424,350 @@ a
...
+    <Sub />
+    <Sub />
+    <Sub />
+  </Element2>
+);
+
+=====================================output=====================================
+$var = ($number % 10 >= 2 && ($number % 100 < 10 || $number % 100 >= 20)
+      ? kochabCooieGameOnOboleUnweave
+      : annularCooeedSplicesWalksWayWay
```
Comment (thorn0) [inline tests/ternaries/__snapshots__/jsfmt.spec.js.snap]: Doesn't look right, too many whitespaces.

**Comment by thorn0** [comment]
There are still some problems with TS conditional types. Don't merge it yet.

**Comment by thorn0** [comment]
Ready for review.

**Comment by evilebottnawi** [comment]
Great job!

**Pull request merged**